language/stdlib/modules/libra_system.move

```diff
@@ -151,11 +175,20 @@ module LibraSystem {
         Vector::borrow(&get_validator_set().validators, i).addr
     }
 
-    // Private helper functions
+    ///////////////////////////////////////////////////////////////////////////
+    // Private functions
+    ///////////////////////////////////////////////////////////////////////////
 
-    // The Association, the VM and the validator operator from the current validator set
-    // are authorized to update the set of validator infos
-    fun is_sender_authorized_(): bool {
+    fun is_valid_and_certified(addr: address): bool {
```

Comment:
Sure! I'd also be happy to discuss this more if you find it should be done in some better way!
But currently:
* "valid" means that the validator has the config set to Option::some (because initially when the association creates an account for the validator, it's config is set to Option::none), so it's up to the validator to make itself valid,
* "certified" means that the association assigned a role to this account (when it calls LibraAccount::certify it sets the is_certified flag of the role to true, when it calls LibraAccount::decertify it sets the is_certified flag of the role to false).

So in order to be added to the ValidatorSet, the validator needs to be certified and valid. Also when LibraSystem::update_and_reconfigure is called, validators who are not valid_and_certified are kicked out of the set.

Reply:
Thanks, that makes sense!

Reply:
Comment pending?